app/domain/sac_imports/wso21_person_entry.rb

```diff
@@ -0,0 +1,167 @@
+
+    def email_matches
+      if person.persisted? && row[:email] != person.email
+        errors.add(:email, "#{row[:email]} does not match the current email")
+      end
+      if !person.persisted? && Person.exists?(email: row[:email])
+        errors.add(:email, "#{row[:email]} already exists")
+      end
+    end
+
+    def at_least_one_role
+      if !person.persisted? && person.roles.empty?
+        errors.add(:roles, "can't be empty")
+      end
+    end
+
+    def valid_gender
+      if !GENDERS.include?(row[:gender].to_sym)
+        errors.add(:gender, "#{row[:gender]} is not a valid gender")
+      end
+    end
+
+    attr_reader :row
+
+    def initialize(row, basic_login_group, abo_group)
+      @row = row
+      @basic_login_group = basic_login_group
+      @abo_group = abo_group
+    end
+
+    def person
+      @person ||= find_or_initialize_person(row).tap do |person|
+        if person.persisted?
+          assign_attributes_for_existing_person(person)
+        else
+          assign_attributes(person)
+        end
+        add_roles(person)
+      end
+    end
+
+    def valid?
+      self_valid = super
+      person_valid = person.valid?
+
+      errors.merge!(person.errors)
+      person.roles.each do |role|
+        errors.merge!(role.errors)
+      end
+
+      self_valid && person_valid
+    end
+
+    def import!
+      raise ActiveRecord::RecordInvalid if !valid?
+      person.save!
+    end
+
+    private
+
+    def assign_common_attributes(person)
+      person.wso2_legacy_password_hash = row[:wso2_legacy_password_hash]
+      person.wso2_legacy_password_salt = row[:wso2_legacy_password_salt]
+      if row[:email_verified] == "1"
+        person.confirmed_at = Time.zone.at(0)
+        person.correspondence = "digital"
+      end
+    end
+
+    def assign_attributes_for_existing_person(person)
+      assign_common_attributes(person)
+    end
+
+    def assign_attributes(person) # rubocop:disable Metrics/AbcSize
+      assign_common_attributes(person)
+
+      person.primary_group = @group
+      if Person.exists?(email: row[:email])
+        person.errors.add(:email, "already exists")
+      end
+      person.email = row[:email]
+
+      person.first_name = row[:first_name]
+      person.last_name = row[:last_name]
+      person.address_care_of = row[:address_care_of]
+      person.postbox = row[:postbox]
+      person.address = row[:address]
+
+      person.street, person.housenumber = row[:address] && Address::Parser.new(row[:address]).parse
+
+      person.country = country
+      person.town = row[:town]
+      person.zip_code = row[:zip_code]
+      person.birthday = row[:birthday]
+      person.gender = gender
+      person.language = language
+      person.phone_numbers.build(
+        number: row[:phone],
+        label: "Mobil"
+      )
+      person.phone_numbers.build(
+        number: row[:phone_business],
+        label: "Arbeit"
+      )
+    end
+
+    def add_roles(person)
```

Comment:
```suggestion
    def assign_roles(person)
```

müsste man ggf. die best Rollen auch ersetzen? wenn das script mehrfach läuft?
personen die ggf. bereits in navision in der `Navision Import` Gruppe auf dem Top Layer sind müssten noch aus dieser Gruppe entfernt werden. Siehe membership import

Reply:
Wird mit SAC/CAS besprochen und dann in einem follow up umgesetzt.